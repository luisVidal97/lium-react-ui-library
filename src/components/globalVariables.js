
const forms = ['contained', 'outlined'];

const heights = {
  large: '50px',
  medium: '40px',
  small: '30px',
}

const borderRadius = {
  strong: '20px',
  normal: '10px',
  none: '0px'
}

const heightsInputText = {
  large: {
    shape: '40px',
    font: '1.2rem'
  },
  medium:  {
    shape: '32px',
    font: '1rem'
  },
  small: {
    shape: '24px',
    font: '.8rem'
  },
}


const colors = {
  primary: {
      normal: '#115293',
      hover: '#0A0A36',
  },
  secondary: {
    normal: '#DC004E',
    hover: '#9A0036',
  },
  danger: {
      normal: '#ff0000',
      hover: '#cc0000',
  },
  warning: {
    normal: '#ffea05',
    hover: '#e2d006'
  },
  success: {
    normal: '#06c920',
    hover: '#019e16'
  },
  dark: {
    normal: '#000000',
    hover: '#686767',
  },
};

const defaultStyle = {
  normal: '#000000',
  hover: '#686767',
}

export { forms, borderRadius, heights, heightsInputText, colors, defaultStyle } ;
