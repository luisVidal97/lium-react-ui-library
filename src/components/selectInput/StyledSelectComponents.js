import styled from 'styled-components';


export const StyledDivTitle = styled.div`
  display: flex;
  justify-content: space-between;
  cursor: pointer;
  border: 1px solid #cccccc;
  border-radius: 5px;
  padding: 5px;
`;

export const StyledTitle = styled.p`
  margin: 0px;
  margin-right: 10px;
`;

export const StyledList = styled.ul`
  list-style: none;
  background-color: white;
  margin: 0;
  padding: 0;
  position: absolute;
  z-index: 100;
  box-shadow: 0 .125rem .25rem rgba(0,0,0,.3) !important;
  top: 50px;
  width: 100%;
  max-height: 300px;
  overflow: auto;
`;


export const StyledButton = styled.button`
  display: flex;
  background-color: ${props => props.color};
  border: 0;
  border-bottom: 1px solid #ccc;
  border-left: 1px solid #ccc;
  border-right: 1px solid #ccc;

  justify-content: space-between;
  padding: 15px 20px 15px 20px;
  text-align: left;
  width: 100%;

  &:hover, &:focus {
    background-color: #ccc;
    cursor: pointer;
    font-weight: bold;
  }
  &:focus {
    background-color: ${props => props.color};
  }
`;

