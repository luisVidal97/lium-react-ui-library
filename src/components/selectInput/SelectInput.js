import React, { useState } from 'react';
import PropTypes from 'prop-types';
import {StyledDivTitle, StyledTitle, StyledList, StyledButton} from './StyledSelectComponents';


const SelectInput = ({title, items, multiSelect = false, onSelectItems}) => {

  const [open, setOpen] = useState(false);
  const [selection, setSelection] = useState([]);
  const showItems = () =>{
    setOpen(!open);
  }

  const handleOnClickSelect = (item)=> {
    if (selection.some(element => element.id === item.id)) {
      let array = [...selection];
      array = array.filter(
        element => element.id !== item.id
      );
      setSelection([...array]);
    } else {
      if (multiSelect) {
        setSelection([...selection, item]);
      } else {
        setSelection([item]);
      }
    }
  }

  const isSleected = (Iditem) => {
    if(selection.some(element => element.id ===Iditem)){
      return '#D1E9FF';
    }else {
      return 'white';
    }

  }



  return (
    <div style={{position:'relative', margin: '10px'}} >

      <StyledDivTitle
        onClick={()=> {
          onSelectItems(selection);
          showItems();
        }}
      >
        <StyledTitle>{title}</StyledTitle>

        <img
          height="25px"
          width="25px"
          src="data:image/svg+xml;base64,PHN2ZyB4bWxucz0iaHR0cDovL3d3dy53My5vcmcvMjAwMC9zdmciIHZlcnNpb249IjEuMSIgeG1sbnM6eGxpbms9Imh0dHA6Ly93d3cudzMub3JnLzE5OTkveGxpbmsiIHhtbG5zOnN2Z2pzPSJodHRwOi8vc3ZnanMuY29tL3N2Z2pzIiB3aWR0aD0iNTEyIiBoZWlnaHQ9IjUxMiIgeD0iMCIgeT0iMCIgdmlld0JveD0iMCAwIDIxMy4zMzMgMjEzLjMzMyIgc3R5bGU9ImVuYWJsZS1iYWNrZ3JvdW5kOm5ldyAwIDAgNTEyIDUxMiIgeG1sOnNwYWNlPSJwcmVzZXJ2ZSIgY2xhc3M9IiI+PGc+CjxnIHhtbG5zPSJodHRwOi8vd3d3LnczLm9yZy8yMDAwL3N2ZyI+Cgk8Zz4KCQk8cG9seWdvbiBwb2ludHM9IjAsNTMuMzMzIDEwNi42NjcsMTYwIDIxMy4zMzMsNTMuMzMzICAgIiBmaWxsPSIjY2NjY2NjIiBkYXRhLW9yaWdpbmFsPSIjMDAwMDAwIiBzdHlsZT0iIiBjbGFzcz0iIj48L3BvbHlnb24+Cgk8L2c+CjwvZz4KPGcgeG1sbnM9Imh0dHA6Ly93d3cudzMub3JnLzIwMDAvc3ZnIj4KPC9nPgo8ZyB4bWxucz0iaHR0cDovL3d3dy53My5vcmcvMjAwMC9zdmciPgo8L2c+CjxnIHhtbG5zPSJodHRwOi8vd3d3LnczLm9yZy8yMDAwL3N2ZyI+CjwvZz4KPGcgeG1sbnM9Imh0dHA6Ly93d3cudzMub3JnLzIwMDAvc3ZnIj4KPC9nPgo8ZyB4bWxucz0iaHR0cDovL3d3dy53My5vcmcvMjAwMC9zdmciPgo8L2c+CjxnIHhtbG5zPSJodHRwOi8vd3d3LnczLm9yZy8yMDAwL3N2ZyI+CjwvZz4KPGcgeG1sbnM9Imh0dHA6Ly93d3cudzMub3JnLzIwMDAvc3ZnIj4KPC9nPgo8ZyB4bWxucz0iaHR0cDovL3d3dy53My5vcmcvMjAwMC9zdmciPgo8L2c+CjxnIHhtbG5zPSJodHRwOi8vd3d3LnczLm9yZy8yMDAwL3N2ZyI+CjwvZz4KPGcgeG1sbnM9Imh0dHA6Ly93d3cudzMub3JnLzIwMDAvc3ZnIj4KPC9nPgo8ZyB4bWxucz0iaHR0cDovL3d3dy53My5vcmcvMjAwMC9zdmciPgo8L2c+CjxnIHhtbG5zPSJodHRwOi8vd3d3LnczLm9yZy8yMDAwL3N2ZyI+CjwvZz4KPGcgeG1sbnM9Imh0dHA6Ly93d3cudzMub3JnLzIwMDAvc3ZnIj4KPC9nPgo8ZyB4bWxucz0iaHR0cDovL3d3dy53My5vcmcvMjAwMC9zdmciPgo8L2c+CjxnIHhtbG5zPSJodHRwOi8vd3d3LnczLm9yZy8yMDAwL3N2ZyI+CjwvZz4KPC9nPjwvc3ZnPg=="
        />
      </StyledDivTitle>
        { open &&
          (
            <StyledList >
              {
                items.map( item => {
                  const selectedColor = isSleected(item.id);
                  return (
                    <li key={item.id}>
                      <StyledButton onClick={() => handleOnClickSelect(item)} color={selectedColor}>
                        <span>{item.value}</span>
                      </StyledButton>
                    </li>);
                })
              }
            </StyledList>
          )
        }

    </div>
  );
}


SelectInput.propTypes = {
  title: PropTypes.string,
  items: PropTypes.arrayOf(PropTypes.exact({
    id: PropTypes.string,
    value: PropTypes.string
  })).isRequired,
  multiSelect: PropTypes.bool,
  onSelectItems: PropTypes.func.isRequired,
}


export default SelectInput;
