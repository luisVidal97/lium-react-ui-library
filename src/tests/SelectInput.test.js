import React from 'react';
import { shallow, mount } from'enzyme';

import 'jest-styled-components';
import renderer from 'react-test-renderer';

import { heightsInputText, colors, borderRadius } from '../components/globalVariables';
import SelectInput from '../components/selectInput/SelectInput';
import {StyledTitle, StyledDivTitle, StyledList, StyledButton, } from '../components/selectInput/StyledSelectComponents';


describe('Tests over <SelectInput/>: features and behavior', () => {

  let wrapper, tree;
  const title = 'Select Exchange';
  const list = [
    { id: '1', value: 'EUR'},
    { id: '2', value: 'LBP'},
    { id: '3', value: 'CAN'},
    { id: '4', value: 'USD'},
  ];


  let valueTest;
  const onSelectItems = (selected) => {
    valueTest = selected;
  };

  test('should show <SelectInput/> successfully', () => {
    wrapper = shallow(<SelectInput items={list} onSelectItems={onSelectItems}/>);
    expect( wrapper ).toMatchSnapshot();
  });



  test('should show select with title and the items', () => {

    wrapper = mount(<SelectInput items={list} onSelectItems={onSelectItems} title={title}/>);

    const titleSelect  = wrapper.find(StyledTitle).text();
    expect(titleSelect).toBe(title);

    wrapper.find(StyledDivTitle).simulate('click');

    const option1 = wrapper.find(StyledButton).at(0).text();
    const option2 = wrapper.find(StyledButton).at(1).text();
    const option3 = wrapper.find(StyledButton).at(2).text();
    const option4 = wrapper.find(StyledButton).at(3).text();

    expect(option1).toBe(list[0].value);
    expect(option2).toBe(list[1].value);
    expect(option3).toBe(list[2].value);
    expect(option4).toBe(list[3].value);

    wrapper.find(StyledButton).at(0).simulate('click');

    wrapper.find(StyledDivTitle).simulate('click');

  });


  test('should select one item and return a simple element in onSelectInput', () => {

    wrapper = mount(<SelectInput items={list} onSelectItems={onSelectItems} title={title}/>);
    wrapper.find(StyledDivTitle).simulate('click');
    wrapper.find(StyledButton).at(0).simulate('click');
    wrapper.find(StyledDivTitle).simulate('click');
    expect(valueTest).toEqual( [ { id: '1', value: 'EUR' } ]);

    wrapper.find(StyledDivTitle).simulate('click');
    wrapper.find(StyledButton).at(2).simulate('click');
    wrapper.find(StyledDivTitle).simulate('click');
    expect(valueTest).toEqual([ { id: '3', value: 'CAN'} ]);

  });


  test('should select several items and return a array of selected elements in onSelectInput', () => {

    wrapper = mount(<SelectInput items={list} onSelectItems={onSelectItems} title={title} multiSelect={true}/>);
    wrapper.find(StyledDivTitle).simulate('click');
    wrapper.find(StyledButton).at(0).simulate('click');
    wrapper.find(StyledButton).at(2).simulate('click');
    wrapper.find(StyledButton).at(3).simulate('click');
    wrapper.find(StyledDivTitle).simulate('click');
    expect(valueTest).toEqual([{ id: '1', value: 'EUR'}, { id: '3', value: 'CAN'}, { id: '4', value: 'USD'}]);

  });

});
