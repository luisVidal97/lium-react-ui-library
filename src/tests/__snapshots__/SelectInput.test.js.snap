// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests over <SelectInput/>: features and behavior should show <SelectInput/> successfully 1`] = `
<div
  style={
    Object {
      "margin": "10px",
      "position": "relative",
    }
  }
>
  <styled.div
    onClick={[Function]}
  >
    <styled.p />
    <img
      height="25px"
      src="data:image/svg+xml;base64,PHN2ZyB4bWxucz0iaHR0cDovL3d3dy53My5vcmcvMjAwMC9zdmciIHZlcnNpb249IjEuMSIgeG1sbnM6eGxpbms9Imh0dHA6Ly93d3cudzMub3JnLzE5OTkveGxpbmsiIHhtbG5zOnN2Z2pzPSJodHRwOi8vc3ZnanMuY29tL3N2Z2pzIiB3aWR0aD0iNTEyIiBoZWlnaHQ9IjUxMiIgeD0iMCIgeT0iMCIgdmlld0JveD0iMCAwIDIxMy4zMzMgMjEzLjMzMyIgc3R5bGU9ImVuYWJsZS1iYWNrZ3JvdW5kOm5ldyAwIDAgNTEyIDUxMiIgeG1sOnNwYWNlPSJwcmVzZXJ2ZSIgY2xhc3M9IiI+PGc+CjxnIHhtbG5zPSJodHRwOi8vd3d3LnczLm9yZy8yMDAwL3N2ZyI+Cgk8Zz4KCQk8cG9seWdvbiBwb2ludHM9IjAsNTMuMzMzIDEwNi42NjcsMTYwIDIxMy4zMzMsNTMuMzMzICAgIiBmaWxsPSIjY2NjY2NjIiBkYXRhLW9yaWdpbmFsPSIjMDAwMDAwIiBzdHlsZT0iIiBjbGFzcz0iIj48L3BvbHlnb24+Cgk8L2c+CjwvZz4KPGcgeG1sbnM9Imh0dHA6Ly93d3cudzMub3JnLzIwMDAvc3ZnIj4KPC9nPgo8ZyB4bWxucz0iaHR0cDovL3d3dy53My5vcmcvMjAwMC9zdmciPgo8L2c+CjxnIHhtbG5zPSJodHRwOi8vd3d3LnczLm9yZy8yMDAwL3N2ZyI+CjwvZz4KPGcgeG1sbnM9Imh0dHA6Ly93d3cudzMub3JnLzIwMDAvc3ZnIj4KPC9nPgo8ZyB4bWxucz0iaHR0cDovL3d3dy53My5vcmcvMjAwMC9zdmciPgo8L2c+CjxnIHhtbG5zPSJodHRwOi8vd3d3LnczLm9yZy8yMDAwL3N2ZyI+CjwvZz4KPGcgeG1sbnM9Imh0dHA6Ly93d3cudzMub3JnLzIwMDAvc3ZnIj4KPC9nPgo8ZyB4bWxucz0iaHR0cDovL3d3dy53My5vcmcvMjAwMC9zdmciPgo8L2c+CjxnIHhtbG5zPSJodHRwOi8vd3d3LnczLm9yZy8yMDAwL3N2ZyI+CjwvZz4KPGcgeG1sbnM9Imh0dHA6Ly93d3cudzMub3JnLzIwMDAvc3ZnIj4KPC9nPgo8ZyB4bWxucz0iaHR0cDovL3d3dy53My5vcmcvMjAwMC9zdmciPgo8L2c+CjxnIHhtbG5zPSJodHRwOi8vd3d3LnczLm9yZy8yMDAwL3N2ZyI+CjwvZz4KPGcgeG1sbnM9Imh0dHA6Ly93d3cudzMub3JnLzIwMDAvc3ZnIj4KPC9nPgo8ZyB4bWxucz0iaHR0cDovL3d3dy53My5vcmcvMjAwMC9zdmciPgo8L2c+CjxnIHhtbG5zPSJodHRwOi8vd3d3LnczLm9yZy8yMDAwL3N2ZyI+CjwvZz4KPC9nPjwvc3ZnPg=="
      width="25px"
    />
  </styled.div>
</div>
`;
